org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/AbstractProjectChangeListener.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * SonarLint for Eclipse
+ * Copyright (C) 2015-2023 SonarSource SA
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarlint.eclipse.ui.internal;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResourceChangeEvent;
+import org.eclipse.core.resources.IResourceChangeListener;
+import org.eclipse.core.resources.IWorkspace;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.jdt.annotation.Nullable;
+
+public abstract class AbstractProjectChangeListener implements IResourceChangeListener {
+  protected IWorkspace workspace;
+  @Nullable
+  protected IProjectChangeHandler handler;
+  
+  protected AbstractProjectChangeListener(IProjectChangeHandler handler) {
+    this.workspace = ResourcesPlugin.getWorkspace();
+    this.handler = handler;
+  }
+  
+  @Override
+  public void resourceChanged(IResourceChangeEvent event) {
+    var rsrc = event.getResource();
+    if (rsrc instanceof IProject) {
```

Comment:
We should avoid relying on IProject (think about our Cobol friends). Instead, you can try to adapt the resource to an ISonarLint project. If it adapts, then work with this ISonarLintProject later on.

Reply:
I removed the project entirely as it does not work with `ISonarLintProject`